File: frontend/src/microservices/blog/api/CreateBlogPostPlanner.ts
// 本文件定义 CreateBlogPostPlanner，负责博客域对应接口入口。

import type { ContentImagePlannerResponse } from '@/microservices/content/objects/ContentImagePlannerResponse'
import type { BlogPostResponse } from '@/microservices/blog/objects/BlogPostResponse'
import type { BlogTagResponse } from '@/microservices/blog/objects/BlogTagResponse'
import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'

export const createBlogPost = (payload: {
  userId: string
  title: string
  summary: string
  content: string
  images: ContentImagePlannerResponse[]
  tags?: BlogTagResponse[]
  coverText?: string
  travelCity?: string | null
  travelCities?: string[] | null
}): Promise<BlogPostResponse> =>
  executeJsonApiRequest('/CreateBlogPostPlanner', 'POST', payload)

File: backend/src/main/scala/microservices/blog/api/CreateBlogPostPlanner.scala
// CreateBlogPostPlanner：博客域创建博客文章入口。

package com.typesafe.travel.blog.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.persistence.blog.BlogPlannerPlainSql

import java.sql.Connection
import java.time.Instant

object CreateBlogPostPlanner extends ConnectionApiPlan[CreateBlogPostPlannerRequest, BlogPostResponse]:
  override val name: String = "CreateBlogPostPlanner"
  override def plan(input: CreateBlogPostPlannerRequest, connection: Connection): IO[BlogPostResponse] =
    val draft = SaveBlogDraftPlannerRequest(
      userId = input.userId,
      postId = None,
      title = input.title,
      summary = input.summary,
      coverText = input.coverText.getOrElse(input.summary),
      content = input.content,
      images = input.images,
      tags = input.tags.getOrElse(Nil),
      travelCity = input.travelCity,
      travelCities = input.travelCities
    )
    validateBlogDraft(draft, requireTags = true) *>
      BlogPlannerPlainSql
        .insertPost(connection, draft, BlogPostStatus.Published.toString, Instant.now())
        .flatMap(postId => BlogPlannerPlainSql.findPost(connection, postId, Some(input.userId)))
